src/mono/mono/metadata/bundled-resources-internals.h

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+
+#ifndef __MONO_METADATA_BUNDLED_RESOURCES_INTERNALS_H__
+#define __MONO_METADATA_BUNDLED_RESOURCES_INTERNALS_H__
+
+#include "loader-internals.h"
+
+#include <mono/metadata/assembly.h>
+#include <mono/metadata/mono-private-unstable.h>
+
+typedef enum {
+	MONO_BUNDLED_DATA,
+	MONO_BUNDLED_ASSEMBLY,
+	MONO_BUNDLED_SATELLITE_ASSEMBLY,
+	MONO_BUNDLED_RESOURCE_COUNT,
+} MonoBundledResourceType;
+
+typedef struct _MonoBundledResource {
+	MonoBundledResourceType type;
+	void (*free_bundled_resource_func)(void *);
+} MonoBundledResource;
+
+typedef struct _MonoBundledData {
+	char *name;
+	const unsigned char *data;
+	unsigned int size;
+} MonoBundledData;
+
+typedef struct _MonoBundledDataResource {
+	MonoBundledResource resource;
+	MonoBundledData data;
+} MonoBundledDataResource;
+
+typedef struct _MonoBundledSymfile {
+	const unsigned char *data;
+	unsigned int size;
+} MonoBundledSymfile;
+
+typedef struct _MonoBundledAssemblyResource {
+	MonoBundledResource resource;
+	MonoBundledAssembly assembly;
+    MonoBundledSymfile symfile;
+} MonoBundledAssemblyResource;
+
+typedef struct _MonoBundledSatelliteAssemblyResource {
+	MonoBundledResource resource;
+	MonoBundledSatelliteAssembly satellite_assembly;
+    MonoBundledSymfile symfile;
+} MonoBundledSatelliteAssemblyResource;
+
+void
+mono_bundled_resources_free (void);
```

Comment:
Having the prefix mono_bundled_resource as well as repeating the same thing in the rest of the function names cause unnecessary long file names. We also mix mono_bundeled_resources and mono_bundled_resource, maybe we should simplify it into:

mono_bundled_resources_free
mono_bundled_resources_add
mono_bundled_resources_get
mono_bundled_resources_contains_assemblies
mono_bundled_resources_contains_satellite_assemblies

Reply:
I wasn't sure if being more explicit about the function name was preferred, but the names have gotten a bit wordy.
I was also wondering if the last two should also include `get_` as similarly suggested here https://github.com/dotnet/runtime/pull/84191#discussion_r1175773920, as they are a getter for the boolean values.

Updated the names to the suggested above